vectordb_orm/backends/pinecone/pinecone.py

```diff
@@ -0,0 +1,256 @@
+import pinecone
+
+from vectordb_orm.attributes import AttributeCompare
+from vectordb_orm.backends.base import BackendBase
+from vectordb_orm.backends.pinecone.indexes import PineconeIndex
+from vectordb_orm.base import VectorSchemaBase
+from vectordb_orm.fields import EmbeddingField
+from vectordb_orm.results import QueryResult
+from vectordb_orm.attributes import OperationType
+
+
+class PineconeBackend(BackendBase):
+    max_fetch_size = 1000
+
+    def __init__(
+        self,
+        api_key: str,
+        environment: str
+    ):
+        # Pinecone can't accept quotes in either parameters, and these are sometimes
+        # unintentionally included in env variables
+        # We quit before initializing with a more informative error message
+        if "'" in api_key or '"' in api_key:
+            raise ValueError("Pinecone `api_key` contains single or double quotes, which isn't allowed.")
+        if "'" in environment or '"' in environment:
+            raise ValueError("Pinecone `environment` contains single or double quotes, which isn't allowed.")
+
+        pinecone.init(
+            api_key=api_key,
+            environment=environment,
+        )
+
+    def create_collection(self, schema: Type[VectorSchemaBase]):
+        collection_name = self.transform_collection_name(schema.collection_name())
+        current_indexes = pinecone.list_indexes()
+        if collection_name in current_indexes:
+            info("Collection already created...")
+            return
+
+        info("Creating collection, this could take 30s-5min...")
+
+        self._assert_valid_collection_name(collection_name)
+        self._assert_has_primary(schema)
+        self._assert_valid_embedding_field(schema)
+
+        # Pinecone allows for dynamic keys on each object
+        # However we need to pre-provide the keys we want to search on
+        # This does increase memory size so we can consider adding an explict index flag
+        # for the metadata fields that we want to allow search on
+        # https://docs.pinecone.io/docs/manage-indexes#selective-metadata-indexing
+        metadata_config = {
+            "indexed": [
+                key
+                for key in schema.__annotations__.keys()
+            ]
+        }
+
+        _, embedding_field = self._get_embedding_field(schema)
+        index : PineconeIndex = embedding_field.index
+
+        pinecone.create_index(
+            name=collection_name,
+            dimension=embedding_field.dim,
+            metric=index.metric_type.value,
+            metadata_config=metadata_config
+        )
+        return pinecone.Index(index_name=collection_name)
+
+    def clear_collection(self, schema: Type[VectorSchemaBase]):
+        collection_name = self.transform_collection_name(schema.collection_name())
+
+        current_indexes = pinecone.list_indexes()
+        if collection_name not in current_indexes:
+            return
+
+        index = pinecone.Index(index_name=collection_name)
```

Comment:
can you describe the scenario in which this is desirable behavior? It's odd that clearing a collection would require first creating an index. Seems like defensive programming and a code smell